app/src/main/java/com/socketmobile/stockcount/ui/EditActivity.kt

```diff
@@ -254,10 +302,17 @@ class EditActivity : AppCompatActivity() {
     }
     @Subscribe(threadMode = ThreadMode.MAIN, sticky = true)
     fun onCaptureDeviceStateChange(event: DeviceStateEvent) {
-        val scannerStatus = event.state.intValue()
-        val deviceGuid = event.device.deviceGuid
-        deviceStateMap[deviceGuid] = event.state
-        deviceClientMap[deviceGuid] = event.device
+        val device = event.device
+        var state = event.state
+        val scannerStatus = state.intValue()
+        val deviceGuid = device.deviceGuid
+        deviceStateMap[deviceGuid] = state
+        deviceClientMap[deviceGuid] = device
+
+        if (!isSocketCamDevice(device.deviceName)) {
+            stopSocketCamExtension()
```

Comment:
When you stop the extension, do you receive a device removal event for the SocketCam reader?

Reply:
yes, receive device removal event after stop the extension